### ISSUE TITLE
Recording get skinny and distorted pixels

### ISSUE THREAD

**Issue by saman1832** [sskodje/ScreenRecorderLib#326]
When starting the application recording on resolution 500*500  then resizing the application then resolution gets distorted.

I tried with this code but no success
 private void Window_SizeChanged(object sender, SizeChangedEventArgs e)
 {
     var a = e.NewSize;
     ScreenSize screenSize = new ScreenSize(a.Width, a.Height);
     if (recorder?.Status == RecorderStatus.Recording)
     {
         recorder?.GetDynamicOptionsBuilder()
             .SetDynamicOutputOptions(new DynamicOutputOptions { VideoFramePreviewSize = screenSize })
             .Apply();
     }
 }
 
 And this is my options:
 IVideoEncoder videoEncoder = new H265VideoEncoder { BitrateMode = H265BitrateControlMode.Quality, EncoderProfile = H265Profile.Main };
 //IntPtr windowHandle = new System.Windows.Interop.WindowInteropHelper(Application.Current.MainWindow).Handle;
 //sources.Add(new WindowRecordingSource(windowHandle));
 ScreenSize screenSize = new ScreenSize(1920, 1080);
 RecorderOptions options = new RecorderOptions
 {
     SourceOptions = new SourceOptions
     {
         RecordingSources = sources
    
     },
     VideoEncoderOptions = new VideoEncoderOptions
     {
         Framerate = 10,
         Encoder= videoEncoder,
         IsFixedFramerate = false,
         IsHardwareEncodingEnabled = true,
         Quality = 70,
     },
     OutputOptions = new OutputOptions
     {
         RecorderMode = RecorderMode.Video,
         OutputFrameSize = screenSize,
         Stretch = StretchMode.Uniform,
         VideoFramePreviewSize = new ScreenSize(0, 150)
     }

 };
 
 @sskodje  pls help me in this as I am stuck

**Comment by sskodje** [comment]
Can you explain a bit more what the problem is and what the expected result is? A sample video of the problem would help.

```
private void Window_SizeChanged(object sender, SizeChangedEventArgs e)
{
var a = e.NewSize;
ScreenSize screenSize = new ScreenSize(a.Width, a.Height);
if (recorder?.Status == RecorderStatus.Recording)
{
recorder?.GetDynamicOptionsBuilder()
.SetDynamicOutputOptions(new DynamicOutputOptions { VideoFramePreviewSize = screenSize })
.Apply();
}
}
```

This alters the size of the frame preview thumbnails you get via OnFrameRecorded callback. Is that the intention?

**Comment by saman1832** [comment]
https://github.com/user-attachments/assets/e8ac6d41-b6cb-4c6c-bfc6-7241b49f1972

**Comment by saman1832** [comment]
@sskodje  I want if i resize the application while recording, or increases the height and width the pixels should not get distorted

**Comment by saman1832** [comment]
@sskodje  any update on this?

**Comment by saman1832** [comment]
@sskodje any way to fix this. As user can start the recording in restore mode and then maximize the window.
this will make the recording resolution minimal.

**Comment by sskodje** [comment]
Yes, but the recording resolution cannot be changed after you have started a recording. You can change the resolution of the source with the OutputSize property on a recording source to match your monitor, and control the scaling with the AnchorPoint, Stretch and Position properties.

**Comment by sskodje** [comment]
Yes, you can do that using Windows Graphics Capture API for now. You save a reference to the RecordingSource you passed to the recorder initially, then change the properties on it, like for example OutputSize, then call

```csharp
            recorderInstance
                .GetDynamicOptionsBuilder()
                .SetUpdatedRecordingSource(recordingSourceInstance)
                .Apply();
```

**Comment by sskodje** [comment]
I apologize for the inconvenience, i have discovered a bug causing issues with dynamically setting outputsize on a source, which is probably why you had issues with it. I'll try push a fix soon.
